react/src/Attachment/AttachmentError.tsx

```diff
@@ -0,0 +1,78 @@
+import { useMemo } from 'react'
+import { FileRejection } from 'react-dropzone'
+import {
+  Button,
+  Flex,
+  forwardRef,
+  Icon,
+  Stack,
+  Text,
+  VisuallyHidden,
+} from '@chakra-ui/react'
+import { dataAttr } from '@chakra-ui/utils'
+
+import { BxCrossCircle } from '~/icons/BxCrossCircle'
+
+import { getErrorMessage } from './utils/getErrorMessage'
+import { useAttachmentStyles } from './AttachmentContext'
+import { getReadableFileSize } from './utils'
+
+export interface AttachmentErrorProps {
+  fileRejection: FileRejection
+  isDisabled?: boolean
+  isReadOnly?: boolean
+  handleDismiss: (fileRejection: FileRejection) => void
+}
+
+export const AttachmentError = forwardRef<AttachmentErrorProps, 'div'>(
+  ({ fileRejection, isDisabled, isReadOnly, handleDismiss }, ref) => {
+    const { file } = fileRejection
+    const styles = useAttachmentStyles()
+
+    const readableFileSize = useMemo(
+      () => getReadableFileSize(file.size),
+      [file.size],
+    )
+
+    return (
+      <Flex
+        aria-disabled={isDisabled}
+        ref={ref}
+        tabIndex={0}
+        sx={{ ...styles.fileInfoContainer, maxHeight: 'fit-content' }}
+      >
+        <VisuallyHidden>
+          File attached: {file.name} with file size of {readableFileSize}
```

Comment:
a11y: This is not what this component represents, we may want to update the error text here. Maybe 

```suggestion
          Invalid file: {getErrorMessage(fileRejection)}
```